tests/ui/truncate_with_drain.rs

```diff
@@ -0,0 +1,282 @@
+#![allow(unused)]
+#![warn(clippy::truncate_with_drain)]
+
+use std::collections::VecDeque;
+
+fn vec_range() {
+    // Do not lint because iterator is assigned
+    let mut v = vec![1, 2, 3];
+    let iter = v.drain(1..v.len());
+
+    // Do not lint because iterator is used
+    let mut v = vec![1, 2, 3];
+    let n = v.drain(1..v.len()).count();
+
+    // Do not lint because iterator is assigned and used
```

Comment:
Isn't that already fully covered by the "Do not lint because iterator is assigned" test?

Reply:
I think `used` and `assigned` is difference in this case and both need to have a test? Maybe I will remove the `assigned_and_used` test.